config/schemas/validationSchemas.js

```diff
@@ -0,0 +1,64 @@
+const z = require("zod");
+
+const usernameSchema = z.string().min(3).max(25);
+
+const basicProfileSchema = z.object({
+  name: z.string().optional(),
+  bio: z.string().optional(),
+  links: z
+    .array(
+      z.object({
+        name: z.string().optional(),
+        url: z.string().url().optional(),
+        username: z.string().optional(),
+        icon: z.string().optional(),
+      })
+    )
+    .optional(),
+});
+
+const socialSchema = z.object({
+  icon: z.string().optional(),
+  url: z.string().url().optional(),
+});
+
+const linkSchema = z.object({
+  group: z.string().optional(),
+  name: z.string().optional(),
+  url: z.string().url().optional(),
+  icon: z.string().optional(),
+  isPinned: z.boolean().optional(),
+});
+
+const testimonialSchema = z.object({
+  title: z.string().optional(),
+  description: z.string().optional(),
+  date: z.string().optional(),
```

Comment:
Could this be a date validation? We have some issues and we need to be more strict with dates

Reply:
I used `z.coerce.date().optional()` instead.

> The coerce method lets you transform or modify the user input before it's validated against the schema. In this case, it converts the date string to a Date object by passing the input through new Date(input). (source: [Powerful Form Validation With React Hook Form and Zod
](https://catalins.tech/form-validation-with-react-hook-form-zod-typescript/))